samples/net/sockets/echo/README.rst

```diff
@@ -0,0 +1,78 @@
+.. _sockets-echo-sample:
+
+Socket Echo Server
+##################
+
+Overview
+********
+
+The sockets/echo sample application for Zephyr implements an IPv4 TCP echo
+server using a BSD Sockets like API. The purpose of this sample is to show
```

Comment:
The idea is that it's (BSD Sockets)-like API, i.e. an API which is like the "BSD Sockets" API. Whereas BSD Sockets-like API implies BSD (Sockets-like) API, i.e BSD API which is like Sockets. And that's not the intended meaning.

But I actually keep mixing up terminology myself, the best moniker I came up with is "BSD Sockets compatible API", which is used in the main docs: https://www.zephyrproject.org/doc/subsystems/networking/bsd-sockets.html . ("BSD Sockets like API" was an early codename ;-) ). Btw, it's "like" and "compatible", because, well, it's just a small subset of the full BSD Sockets API, so adding an extra word should set the expectations right.

Let me prepare a new patch to apply the naming consistently.